libs/dyn/jsonloader/json_test.go

```diff
@@ -0,0 +1,93 @@
+package jsonloader
+
+import (
+	"testing"
+
+	"github.com/databricks/cli/libs/dyn/convert"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/stretchr/testify/require"
+)
+
+const jsonData = `
+{
+    "job_id": 123,
+    "new_settings": {
+        "name": "xxx",
+        "email_notifications": {
+            "on_start": [],
+            "on_success": [],
+            "on_failure": []
+        },
+        "webhook_notifications": {
+            "on_start": [],
+            "on_failure": []
+        },
+        "notification_settings": {
+            "no_alert_for_skipped_runs": true,
+            "no_alert_for_canceled_runs": true
+        },
+        "timeout_seconds": 0,
+        "max_concurrent_runs": 1,
+        "tasks": [
+            {
+                "task_key": "xxx",
+                "email_notifications": {},
+                "notification_settings": {},
+                "timeout_seconds": 0,
+                "max_retries": 0,
+                "min_retry_interval_millis": 0,
+                "retry_on_timeout": "true"
+            }
+        ]
+    }
+}
+`
+
+func TestJsonLoader(t *testing.T) {
+	v, err := LoadJSON([]byte(jsonData), "(inline)")
+	require.NoError(t, err)
+
+	var r jobs.ResetJob
+	err = convert.ToTyped(&r, v)
+	require.NoError(t, err)
+}
+
+const malformedMap = `
+{
+    "job_id": 123,
+    "new_settings": {
+        "name": "xxx",
+        "wrong",
+    }
+}
+`
+
+func TestJsonLoaderMalformedMap(t *testing.T) {
+	_, err := LoadJSON([]byte(malformedMap), "(inline)")
+	require.ErrorContains(t, err, "error decoding JSON at (inline):6:16: invalid character ',' after object key")
+}
+
+const malformedArray = `
+{
+    "job_id": 123,
+    "new_settings": {
+        "name": "xxx",
+        "tasks": [1, "asd",]
+    }
+}`
+
+func TestJsonLoaderMalformedArray(t *testing.T) {
+	_, err := LoadJSON([]byte(malformedArray), "path/to/file.json")
+	require.ErrorContains(t, err, "error decoding JSON at path/to/file.json:6:28: invalid character ']' looking for beginning of value")
+}
+
+const eofData = `
+{
+    "job_id": 123,
+    "new_settings": {
+        "name": "xxx",`
+
+func TestJsonLoaderEOF(t *testing.T) {
+	_, err := LoadJSON([]byte(eofData), "path/to/file.json")
+	require.ErrorContains(t, err, "unexpected end of JSON input")
+}
```

Comment:
s/require/assert/ for the above tests. It's the last statement so the test does not need to be terminated.